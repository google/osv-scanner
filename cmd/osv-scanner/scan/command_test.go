package scan_test

import (
	"testing"

	"github.com/google/osv-scanner/v2/cmd/osv-scanner/internal/testcmd"
)

// Tests all subcommands here.
func TestCommand_SubCommands(t *testing.T) {
	t.Parallel()

	tests := []testcmd.Case{
		// without subcommands
		{
			Name: "with no subcommand",
			Args: []string{"", "./fixtures/locks-many/composer.lock"},
			Exit: 0,
		},
		// with scan subcommand
		{
			Name: "with scan subcommand",
			Args: []string{"", "scan", "./fixtures/locks-many/composer.lock"},
			Exit: 0,
		},
		// scan with a flag
		{
			Name: "scan with a flag",
			Args: []string{"", "scan", "--recursive", "./fixtures/locks-one-with-nested"},
			Exit: 0,
		},
		// TODO: add tests for other future subcommands
	}

	for _, tt := range tests {
		t.Run(tt.Name, func(t *testing.T) {
			t.Parallel()
			testcmd.RunAndMatchSnapshots(t, tt)
		})
	}
}
