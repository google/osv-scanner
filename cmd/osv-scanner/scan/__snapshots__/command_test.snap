
[TestCommand_SubCommands/scan_with_a_flag - 1]
Scanning dir ../fixtures/locks-one-with-nested
Scanned <rootdir>/osv-scanner/fixtures/locks-one-with-nested/nested/composer.lock file and found 1 package
Scanned <rootdir>/osv-scanner/fixtures/locks-one-with-nested/yarn.lock file and found 1 package
No issues found

---

[TestCommand_SubCommands/scan_with_a_flag - 2]

---

[TestCommand_SubCommands/with_no_subcommand - 1]
Scanning dir ../fixtures/locks-many/composer.lock
Scanned <rootdir>/osv-scanner/fixtures/locks-many/composer.lock file and found 1 package
Loaded filter from: <rootdir>/osv-scanner/fixtures/locks-many/osv-scanner.toml
No issues found

---

[TestCommand_SubCommands/with_no_subcommand - 2]

---

[TestCommand_SubCommands/with_scan_subcommand - 1]
Scanning dir ../fixtures/locks-many/composer.lock
Scanned <rootdir>/osv-scanner/fixtures/locks-many/composer.lock file and found 1 package
Loaded filter from: <rootdir>/osv-scanner/fixtures/locks-many/osv-scanner.toml
No issues found

---

[TestCommand_SubCommands/with_scan_subcommand - 2]

---
